spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -95,6 +104,7 @@
   private Filter[] pushedFilters = NO_FILTERS;
   private final boolean localityPreferred;
   private final int batchSize;
+  private ExecutorService readTasksInitExecutorService = DEFAULT_READTASKS_INIT_EXECUTOR_SERVICE;
```

Comment:
I don't think that there is a need for this field and I'd prefer not to add mutable state. Can you refactor this to call `executeWith(ThreadPools.getWorkerPool())`  instead? You can pass `null` to that method so it can also check `localityPreferred` inline:

```
    .executeWith(localityPreferred ? ThreadPools.getWorkerPool() : null)
```

Reply:
I like this comment because it really simplify the code path without introducing any static or local variables.

Reply:
That really simplifies the code. Thanks @rdblue